include/uv.h

```diff
@@ -1414,9 +1417,35 @@ UV_EXTERN void uv_key_delete(uv_key_t* key);
 UV_EXTERN void* uv_key_get(uv_key_t* key);
 UV_EXTERN void uv_key_set(uv_key_t* key, void* value);
 
+
+/*
+ * CPU masks for thread affinity must be at least UV_CPU_SETSIZE bytes long.
+ */
+#if defined(_WIN32)
+# if defined(_WIN64)
+#  define UV_CPU_SETSIZE        64
+# else
+#  define UV_CPU_SETSIZE        32
+# endif
+#else
+# if (defined(__APPLE__) && defined(__MACH__)) || defined(_AIX)
+#  define UV_CPU_SETSIZE        8
+# else
+#  define UV_CPU_SETSIZE        CPU_SETSIZE
+# endif
+#endif
+
 typedef void (*uv_thread_cb)(void* arg);
 
 UV_EXTERN int uv_thread_create(uv_thread_t* tid, uv_thread_cb entry, void* arg);
+UV_EXTERN int uv_thread_setaffinity(uv_thread_t* tid,
+                                    char *cpumask,
+                                    char *oldmask,
+                                    size_t mask_size);
```

Comment:
I'm not actually certain why the atomicity of the operation matters -- a situation where multiple threads are getting/setting the affinity of another thread seems improbable to me. The way I'm using this functionality is for each thread to figure out its own desired affinity and set it. The ``uv_thread_getaffinity()`` call was actually only added for testing purposes; see the discussion in https://github.com/libuv/libuv/pull/273.